File: oolab/src/main/java/agh/ics/oop/model/MapDirection.java
package agh.ics.oop.model;

public enum MapDirection {
    NORTH, EAST, SOUTH, WEST;;

    @Override
    public String toString() {
        return switch (this){
            case NORTH -> "N";
            case EAST -> "E";
            case WEST -> "W";
            case SOUTH -> "S";
        };
    }

    public MapDirection next(){
        switch (this){
            case NORTH -> {
                return EAST;
            }
            case SOUTH -> {
                return WEST;
            }
            case EAST -> {
                return SOUTH;
            }
            case WEST -> {
                return NORTH;
            }
        }
        return null;
    }

    public MapDirection prev(){
        switch (this){
            case NORTH -> {return WEST;}
            case SOUTH -> {return EAST;}
            case EAST -> {return NORTH;}
            case WEST -> {return SOUTH;}
        }
        return null;
    }

    public Vector2d toUnitVector(){
        switch (this){
            case NORTH -> {return new Vector2d(0,1);}
            case SOUTH -> {return new Vector2d(0,-1);}
            case EAST -> {return new Vector2d(1,0);}
            case WEST -> {return new Vector2d(-1,0);}
        }
        return null;
    }
}

File: kotlin/src/main/kotlin/module/MapDirection.kt
package org.example.module

enum class MapDirection {
    NORTH, EAST, SOUTH, WEST;

    override fun toString(): String {
        return when (this) {
            NORTH -> "N"
            EAST -> "E"
            SOUTH -> "S"
            WEST -> "W"
        }
    }

    fun next(): MapDirection {
        return when (this) {
            NORTH -> EAST
            EAST -> SOUTH
            SOUTH -> WEST
            WEST -> NORTH
        }
    }

    fun prev(): MapDirection {
        return when (this) {
            NORTH -> WEST
            EAST -> NORTH
            SOUTH -> EAST
            WEST -> SOUTH
        }
    }
}